sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala

```diff
@@ -1035,6 +1060,41 @@ class PlanResolutionSuite extends AnalysisTest {
 
         case _ => fail("Expect UpdateTable, but got:\n" + parsed4.treeString)
       }
+
+      parsed5 match {
+        case UpdateTable(
+        AsDataSourceV2Relation(_),
+        Seq(Assignment(name: UnresolvedAttribute, UnresolvedAttribute(Seq("DEFAULT"))),
+        Assignment(age: UnresolvedAttribute, UnresolvedAttribute(Seq("DEFAULT")))),
+        None) =>
+          assert(name.name == "name")
+          assert(age.name == "age")
+
+        case _ => fail("Expect UpdateTable, but got:\n" + parsed5.treeString)
+      }
+
+      parsed6 match {
+        case UpdateTable(
+        AsDataSourceV2Relation(_),
+        Seq(Assignment(i: AttributeReference, AnsiCast(Literal(null, _), IntegerType, _)),
```

Comment:
Hmm, probably I missed this in the previous reviews. So we will insert a NULL value for a default expression if the default column value is not defined, right?

Reply:
Yes: when resolving DEFAULT column references, the analyzer will insert literal NULL values if the corresponding table does not define an explicit default value for that column. This is intended.

I left a comment here to this effect in case it's confusing to any future readers.